src/main/io/gps.c

```diff
@@ -2626,9 +2627,18 @@ void onGpsNewData(void)
 #ifdef USE_GPS_RESCUE
     gpsRescueNewGpsData();
 #endif
+
 #ifdef USE_GPS_LAP_TIMER
     gpsLapTimerNewGpsData();
 #endif // USE_GPS_LAP_TIMER
+
+    newDataForPosHold = true;
+}
+
+bool getIsNewDataForPosHold(void) {
+    bool currentState = newDataForPosHold; // true only when new data arrives
+    newDataForPosHold = false; // clear flag once new data has been handled
+    return currentState;
```

Comment:
`onGpsNewData()` is running whenever new GPS data is available. Just implement a new function `posHoldNewGpsData()` in posHold.c/h and in that function just run whatever you want whenever new GPS data is available. Triggering a flag that then again needs another loop to check this is very inconvenient at the least.
```suggestion

#ifdef USE_POS_HOLD
    posHoldNewGpsData();
#endif // USE_POS_HOLD
}
```

Reply:
Ah, I see, that's a better way to do it. Done.